crates/oxc_language_server/src/main.rs

```diff
@@ -492,21 +544,36 @@ impl Backend {
         if config.exists() {
             config_path = Some(config);
         }
-        if let Some(config_path) = config_path {
-            let mut linter = self.server_linter.write().await;
-            let config = Oxlintrc::from_file(&config_path)
-                .expect("should have initialized linter with new options");
-            let config_store = ConfigStoreBuilder::from_oxlintrc(true, config.clone())
-                .expect("failed to build config")
-                .build()
-                .expect("failed to build config");
-            *linter = ServerLinter::new_with_linter(
-                Linter::new(LintOptions::default(), config_store).with_fix(FixKind::SafeFix),
+
+        let config_path = config_path?;
+        let oxlintrc = Oxlintrc::from_file(&config_path)
+            .expect("should have initialized linter with new options");
+        let config_store = ConfigStoreBuilder::from_oxlintrc(true, oxlintrc.clone())
+            .expect("failed to build config")
+            .build()
+            .expect("failed to build config");
+
+        #[expect(clippy::needless_late_init)]
+        let linter;
+        if self.options.lock().await.use_nested_configs {
+            let nested_configs = self.nested_configs.pin();
+            let mut nested_configs_copy: FxHashMap<PathBuf, ConfigStore> = FxHashMap::default();
+            for (path, config_store) in &nested_configs {
```

Comment:
I haven't been able to get `.extend` to work for this. https://github.com/oxc-project/oxc/pull/9724#discussion_r1992462537

I think I'm just misunderstanding how to use it or maybe the HashMap implementations we are using don't support it somehow.

Reply:
I think I got this figured out now.